.zuul.yaml

```diff
@@ -0,0 +1,22 @@
+- job:
+    name: dnf-run-tests
+    description: Run DNF tests
+    roles:
+      - zuul: zuul-distro-jobs
+    requires: repo
```

Comment:
"require" is part of the "provide/require" mechanic where a job can tells Zuul (I provide an artifact named 'abc') or (I require an artifact 'abc'). It means for that dnf-run-tests job Zuul will look at the its database to find if a "related" job (here the parent job dnf-rpm-build) provides a "repo" artifact. This is the case, so Zuul will expose the link of the parent's job artifact into the dnf-run-tests job's ansible's inventory.

When I say "related" jobs, it also mean artifacts of jobs "connected" via the "Depends-On keyword".

To make it clear:
In the job ansible inventory: https://fedora.softwarefactory-project.io/zuul/build/ad4bac1ffda8462380d34c4e5f2f7829/log/zuul-info/inventory.yaml
See the vars.zuul.artifacts. There are two artifacts named "repo". On for dnf and the other for libdnf (The PR linked via the Depends-On (in this PR description/message). Both links are valid rpm repositories.

The role "repo-setup" install on the test system "/etc/yum.repos.d/zuul-build.repo" populated with those artifacts's repos.

In the job playbook we display the content of that file: https://fedora.softwarefactory-project.io/zuul/build/ad4bac1ffda8462380d34c4e5f2f7829/log/job-output.txt#162

So from there we can fetch the rpms for the repos or whatever we need for our new job.